packages/@aws-cdk/aws-s3-deployment/README.md

```diff
@@ -0,0 +1,68 @@
+## AWS S3 Deployment Construct Library
+
+> __Status: Experimental__
+
+This library allows populating an S3 bucket with the contents of a .zip file
+from another S3 bucket or from local disk.
+
+The following example defines a publicly accessible S3 bucket with web hosting
+enabled and populates it from a local directory on disk.
+
+```ts
+const websiteBucket = new s3.Bucket(this, 'WebsiteBucket', {
+  websiteIndexDocument: 'index.html',
+  publicReadAccess: true
+});
+
+new s3deploy.BucketDeployment(this, 'DeployWebsite', {
+  source: s3deploy.Source.asset('./website-dist'),
+  destinationBucket: websiteBucket,
+  destinationKeyPrefix: 'web/static' // optional prefix in destination bucket
+});
+```
+
+This is what happens under the hood:
+
+1. When this stack is deployed (either via `cdk deploy` or via CI/CD), the
+   contents of the local `website-dist` directory will be archived and uploaded
+   to an intermediary assets bucket.
+2. The `BucketDeployment` construct synthesizes a custom CloudFormation resource
+   of type `Custom::CDKBucketDeployment` into the template. The source bucket/key
+   is set to point to the assets bucket.
+3. The custom resource downloads the .zip archive, extracts it and issues `aws
+   s3 sync --delete` against the destination bucket (in this case
+   `websiteBucket`).
+
+## Supported sources
+
+The following source types are supported for bucket deployments:
+
+ - Local .zip file: `s3deploy.Source.asset('/path/to/local/file.zip')`
+ - Local directory: `s3deploy.Source.asset('/path/to/local/directory')`
+ - Another bucket: `s3deploy.Source.bucket(bucket, zipObjectKey)`
+
+## Retain on Delete
+
+By default, the contents of the destination bucket will be deleted when the
+`BucketDeployment` resource is removed from the stack. You can use the option
+`retainOnDelete: true` to disable this behavior.
+
+## Notes
+
+ * This library uses an AWS CloudFormation custom resource which about 10MiB in
```

Comment:
That is frikkin' horrible. For pragmatic reasons I get that we might not care for now, but we should bring this down.

Reply:
That's the size of AWS CLI. It will be hard to bring this down really.

Reply:
For one thing we could not use the AWS CLI.

Reply:
Find me a reliable implementation of "s3 sync" and I'll change.

Reply:
Heh, fair enough.